concordium-node/src/bin/cli.rs

```diff
@@ -196,38 +200,45 @@ async fn main() -> anyhow::Result<()> {
     // Start baking
     consensus.start_baker();
 
-    // Everything is running, so we wait for a signal to shutdown.
-    if shutdown_signal_1.recv().await.is_err() {
+    // Everything is running, so we wait for a signal to shutdown or for an error to
+    // occur.
+    if shutdown_receiver.recv().await.is_err() {
         error!("Shutdown signal handler was dropped unexpectedly. Shutting down.");
     }
 
+    info!("Initiating shutdown.");
+
     // Message rpc to shutdown first
     if let Some(task) = rpc_server_task {
-        if shutdown_rpc_sender.send(()).is_err() {
-            error!("Could not stop the RPC server correctly. Forcing shutdown.");
-            task.abort();
-        }
-        // Force the rpc server to shut down in at most 10 seconds.
-        let timeout_duration = std::time::Duration::from_secs(10);
-        match tokio::time::timeout(timeout_duration, task).await {
-            Ok(res) => {
-                if let Err(err) = res {
-                    if err.is_cancelled() {
-                        info!("RPC server was successfully shutdown by force.");
-                    } else if err.is_panic() {
-                        error!("RPC server panicked: {}", err);
+        if !task.is_finished() {
```

Comment:
It would be useful to add a short comment about why we do this. That is, in what scenarios the task will already be finished.